### PR TITLE
feat(lint): lint --all

### LINKED ISSUE #313: Would be nice to lint all charts at once
Be able to lint all charts and output json (so that @sgoings can scrape through the travis ci log when travis builds master and update bades to either success/fail) Something like ...
```
[helm-lint] Running helm-lint...
{ "charts": 
  [
    {
      "name": "redis", 
      "status": 50, 
      "failed_criteria": ["lint-type-1", "lint-type-2"]
    }, 
    {
      "name": "alpine", 
      "status": 100
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by lanej**
lint all charts in one shot.  Fixes #313

### CODE REVIEW

**Comment by lanej** [comment]
not done yet.  just a place holder.

**Comment by adamreese** [comment]
Looking good.  Can you break out the unrelated refactors into separate commits?

File: action/helper_test.go [hunk 1]
```diff
@@ -50,3 +52,46 @@ func expect(t *testing.T, a interface{}, b interface{}) {
 		t.Errorf("\n[Expected] type: %v\n%v\n[Got] type: %v\n%v\n", reflect.TypeOf(b), b, reflect.TypeOf(a), a)
 	}
 }
+
+func expectMatches(t *testing.T, actual string, expected string) {
```
Comment (adamreese) [inline action/helper_test.go]: :heart: these helpers

**Comment by adamreese** [comment]
does this pr close any of the other lint issues?

**Comment by lanej** [comment]
Fixes #312

**Comment by technosophos** [comment]
We're going to hit a major conflict with this, since @gerred is doing a similar refactor of the CLI.

Can we split this into separate commits so that we can get the Lint stuff right away?

**Comment by lanej** [comment]
@technosophos how would you recommend splitting them to avoid @gerred's refactor

**Comment by gerred** [comment]
Do what you need, I can work around it and use it as a base if this will be in fast enough.

**Comment by lanej** [comment]
```
./bin/helm lint --all
[ERROR] Missing a label: `heritage: helm` in /Users/deploy/.helm/workspace/charts/alpine/manifests/alpine-pod.yaml
[WARN] Chart [alpine] failed some checks
[ERROR] A README file was not found in /Users/deploy/.helm/workspace/charts/redis
[ERROR] A Chart.yaml file was not found in /Users/deploy/.helm/workspace/charts/redis
[ERROR] A manifests directory was not found in /Users/deploy/.helm/workspace/charts/redis
[ERROR] open /Users/deploy/.helm/workspace/charts/redis/Chart.yaml: no such file or directory
[ERROR] Error walking manifest files. Err: stat /Users/deploy/.helm/workspace/charts/redis/manifests: no such file or directory
[WARN] Chart [redis] failed some checks
Exiting with errors
```

**Comment by technosophos** [comment]
@lanej what @gerred  has been working on is moving that CLI code to a package called `cli` (instead of putting it in `action`). Would you want to just move that code into a new package called `cli` and call it good? Or maybe I missed the nuances behind putting it in `action`.

Other than that decision, this is looking great!

**Comment by gerred** [comment]
Hey @technosophos @lanej leave it in here for now. Once it's merged in, I'll rebase the changes in and get my branch up. So far, I haven't touched `helm.go`, so that'll be the easiest.

**Comment by lanej** [comment]
Fixes #299

**Comment by lanej** [comment]
@gerred did you mean to lift the lgtm2 flag ?

**Comment by gerred** [comment]
Yeah. I'm not sure how this is panning out with #342 - need to coordinate these two.

**Comment by lanej** [comment]
@gerred somebody has to go first.  i don't mind resolving the conflicts.

**Comment by adamreese** [comment]
@gerred Let's make a decision on this.  Rebasing will start becoming an issue with a PR this size.  Anything I can help on?

**Comment by technosophos** [comment]
The CLI refactor needs to be removed from this commit. I'll LGTM it when that's the case.

**Comment by lanej** [comment]
The tests require the CLI refactor.  Some one is going to eat the cost of a rebase.  I don't want to lose viable code in the process.

* We can wait for @gerred's PR to finish and I can rebase on top of that.
* We merge this now and make @gerred do the rebase.
* I merge in @gerred's branch and combine the PRs.

Option number 3 is weird.  I'd rather do 1 or 2.

**Comment by lanej** [comment]
@technosophos ^^

**Comment by gerred** [comment]
@lanej I have some prior commitments preventing me from doing the second bullet in a timely manner (until the end of the week most likely), unfortunately, so the first option may be the best?

**Comment by michelleN** [comment]
Btw, @lanej - I like the way you handled / listed errors here. It'll help with issue #312. :+1:

File: action/action.go [hunk 2]
```diff
@@ -5,28 +5,12 @@ import (
 
 	"github.com/helm/helm/config"
 	"github.com/helm/helm/log"
+	helm "github.com/helm/helm/util"
```
Comment (adamreese) [inline action/action.go]: I'm not a fan of generic `util` packages, but that could be just me.  I'm working though it.

**Comment by adamreese** [comment]
Looks great Josh.  My LGTM stands

**Comment by lanej** [comment]
@technosophos care to weigh in ?

**Comment by technosophos** [comment]
I'm generally not a fan of `util` as a package name, but I wouldn't stop this PR for it. Besides, the way you wrote it, we could just write a regex to change it if we want to later. So I'm adding LGTM2.

**Comment by lanej** [comment]
@technosophos not married to `util`. it was just the easiest.

**Pull request merged**